File: 01-数据结构/01-数组/c++/LC 移除元素.cpp
#include<vector>

using namespace std;

class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int n = nums.size();
        int left = 0;
        int right = 0;

        while(right < n) {
            if (nums[right] != val) {
                swap(nums[left], nums[right]);
                left++;
            }
            right++;
        }
        return left;
    }
};

File: 01-数据结构/01-数组/go/LC 移除元素.go
package main

func removeElement(nums []int, val int) int {
    l, r := 0,0
    for r < len(nums) {
        if nums[r] != val {
            nums[l], nums[r] = nums[r], nums[l]
            l++
        }
        r++
    }
    return l;
}